runtime/cudaq/spin/spin_op.cpp

```diff
@@ -414,9 +414,11 @@ spin_op &spin_op::operator+=(const spin_op &v) noexcept {
 
   for (auto [term, coeff] : tmpv.terms) {
     auto iter = terms.find(term);
-    if (iter != terms.end())
+    if (iter != terms.end()) {
       iter->second += coeff;
-    else
+      if (std::abs(iter->second) < 1e-12)
+        terms.erase(iter);
+    } else
```

Comment:
Even though the `else` block is a simple one liner, let's put braces around it to match the `if then` part. I think that'll make it evre so slightly easier to skim-read.